libvast/src/system/explore_command.cpp

```diff
@@ -0,0 +1,160 @@
+  // Get a local actor to interact with `sys`.
+  caf::scoped_actor self{sys};
+  // TODO: Add --format option to select output format.
+  auto out = detail::make_output_stream(
+    std::string{format::json::writer::defaults::write}, false);
+  if (!out)
+    return make_message(out.error());
+  caf::actor writer
+    = sys.spawn(sink<format::json::writer>,
+                format::json::writer{std::move(*out)}, max_events);
+  if (!writer) {
+    return make_message(ec::unspecified, "could not spawn writer");
+  }
+  auto writer_guard = caf::detail::make_scope_guard(
+    [&] { self->send_exit(writer, caf::exit_reason::user_shutdown); });
+  self->monitor(writer);
+  // Get VAST node.
+  auto node_opt
+    = system::spawn_or_connect_to_node(self, options, content(sys.config()));
+  if (auto err = caf::get_if<caf::error>(&node_opt))
+    return caf::make_message(std::move(*err));
+  auto& node = caf::holds_alternative<caf::actor>(node_opt)
+                 ? caf::get<caf::actor>(node_opt)
+                 : caf::get<scope_linked_actor>(node_opt).get();
+  VAST_ASSERT(node != nullptr);
+  // Start signal monitor.
+  std::thread sig_mon_thread;
+  auto guard = system::signal_monitor::run_guarded(
+    sig_mon_thread, sys, defaults::system::signal_monitoring_interval, self);
+  // Spawn exporter for the passed query
+  auto spawn_exporter
+    = command::invocation{invocation.options, "spawn exporter", {query}};
+  VAST_DEBUG(&invocation, "spawns exporter with parameters:", spawn_exporter);
+  auto exporter = spawn_at_node(self, node, spawn_exporter);
+  if (!exporter)
+    return caf::make_message(std::move(exporter.error()));
+  auto exp_guard = caf::detail::make_scope_guard(
+    [&] { self->send_exit(*exporter, caf::exit_reason::user_shutdown); });
+  // Spawn explorer at the node.
+  auto explorer_options = invocation.options;
+  auto& explore_dict = explorer_options["explore"].as_dictionary();
+  explore_dict["before"] = before;
+  explore_dict["after"] = after;
+  auto spawn_explorer = command::invocation{explorer_options, "spawn explorer"};
+  VAST_DEBUG(&invocation, "spawns explorer with parameters:", spawn_explorer);
+  auto explorer = spawn_at_node(self, node, spawn_explorer);
+  if (!explorer)
+    return caf::make_message(std::move(explorer.error()));
+  auto explorer_guard = caf::detail::make_scope_guard(
+    [&] { self->send_exit(*explorer, caf::exit_reason::user_shutdown); });
+  self->monitor(*explorer);
+  self->send(*explorer, explorer_initial_query_atom::value, *exporter);
+  // Set the explorer as sink for the initial query exporter.
+  self->send(*exporter, system::sink_atom::value, *explorer);
+  // (Ab)use query_statistics as done message.
+  self->send(*exporter, system::statistics_atom::value, *explorer);
+  self->send(*explorer, system::sink_atom::value, writer);
+  self->send(*exporter, system::run_atom::value);
+  caf::error err;
+  auto stop = false;
+  self
```

Comment:
It feels like there should be an opportunity here to deduplicate some of the code that is extremely similar to `pivot_command.cpp` and lift it into common abstractions, but nothing immediately came to mind.

Reply:
Let's not do this in the scope of the initial version. We can always unify the code later on in a follow-up story.